libsolidity/ast/Types.h

```diff
@@ -736,6 +736,13 @@ class ArrayType: public ReferenceType
 
 	TypePointer copyForLocation(DataLocation _location, bool _isPointer) const override;
 
+	/// The offset to advance in calldata to move from one array element to the next.
+	unsigned calldataStride() const { return isByteArray() ? 1 : m_baseType->calldataEncodedSize(); }
+	/// The offset to advance in memory to move from one array element to the next.
+	unsigned memoryStride() const { return isByteArray() ? 1 : m_baseType->memoryHeadSize(); }
+	/// The offset to advance in storage to move from one array element to the next.
```

Comment:
Please add the caveat that this does not take slot boundaries into account.

Reply:
What do you mean by that?

Reply:
If the next element is not in the same storage slot anymore, you need to apply a new logic. This might be obvious, but still..

Reply:
Probably someone should recheck what happens e.g. for arrays of statically sized structs for which the struct (i.e. each array element) is larger than 32-bytes - for those this may in fact be wrong...

Reply:
If so, then ``storageBytes`` is misleading for those already IMO, though...

Reply:
Yep, `storageBytes` is only meaningful if less than 32.

Reply:
In general, it always has to be consulted in combination with `storageSlots`.